### ISSUE TITLE
Návrh features

### ISSUE THREAD

**Issue by jasho** [jasho/fit-project-checker#2]
Do tohto issue by som spisoval features, ktoré nás napadnú, že by sa mohli podporovať.

1. Overenie názvu archívu 
1. Overenie správnej kompresie
1. Overenie obsahu archívu
1. Overenie, že archív obsahuje len to, čo má - podľa názvu
1. Overenie veľkosti - že nie je podozrivo malý ani veľký
1. Overenie maximálne veľkosti podľa nastavenia pri projekte vo WISe
1. Overenie, že typ kompresie sedí s príponou archívu
1. Podpora pre voliteľné názvy súborov a nepovinné rozšírenia (napr. pre IFJ)
1. Kontrola na to, že súbory neobsahujú diakritiku (v niektorých predmetoch je to vyžadované)
1. Kontrola enkódovania súborov
1. Kontrola CR/LF podľa špecifikácií projektu
1. Rozbalenie archívu a skúsenie buildu projektu
1. Overenie, že projekt sa dokáže spustiť na referenčnom prostredí
1. Kontrola správneho formátu hlavičky súboru (napr. pre INP)
1. Kontrola, že súbory neobsahujú nejaké nepovolené slová (napr. xplagiat00)

**Comment by HoundThe** [comment]
Myslím že 2. a poslední feature spolu souvisí - kolidují. Soubor nemá správnou příponu - chyba, Soubor má správnou příponu a špatnou kompresi - chyba, soubor má správnou příponu a správnou kompresi - úspěch.

Do budoucna, co jsem pochopil tak by se dalo vzít archiv, jeho obsah rozbalit do určité složky, zkusit buildnout, jestli se podaří.

**Comment by jasho** [comment]
Súhlasím, že body 2 a 7 spolu súvisia, dokonca bod 7 súvisí aj s bodom 1, ale každý z nich je podľa mňa samostatná chyba.

**Comment by loroka** [comment]
Některé projekty vyžadovaly specifickou hlavičku souboru se jménem datumem atd... V INP si takhle pár lidí kód distribuovalo a zapoměli si změnit jméno z xplagiat00 :chart_with_upwards_trend:. Takže možná by se mohla závést i jistá kontrola textu v souboru? Ovšem ten formát je proměnný a vyžadovalo by to velkou volnost formátu.

**Comment by loroka** [comment]
Upozornit pokud je v archivu soubor který byl modifikován později jak nějaká doba. Hodně lidí nepoužívá verzování, mohou v tom mít bordel a přidat nějakou starší verzi souboru.

**Comment by jasho** [comment]
@loroka veci z tvojho prvého komentára som pridal, to by sa mohlo hodiť.
U toho druhého komentára - môžeš to prosím vysvetliť na nejakom príklade? Z toho popisu som úplne nepochopil o čo sa tam jedná.

**Comment by loroka** [comment]
File system uchovává u souborů informaci o tom, kdy byly naposledy změněny a přidání je do archivu, by tuto informaci nemělo upravit. S předpokladem, že projekt archivuji v momentě, kdy jsem s projektem hotový (tedy ještě jsem soubor před chvílí nějak modifikoval), by se mohlo uživatele upozornit na to, že v má v archivu soubor, který byl modifikován nejpozději dva dny zpět. Dokážu si představit situaci, že někdo si udělá "uložení stavu" projektu překopírováním do jiné složky, takto to udělá víckrát a pak archivuje špatnou složku. Ale uznávám, že je to nachylné na false positive.
